### PR TITLE
Item Panel

### PR DESCRIPTION
**Pull request by PBWiseman**
## Describe your changes

Reworking the item panel to show current stats

## Issue number and link

#104 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by PBWiseman** [comment]
New item panel has been added. It now has a reroll button with 2 charges over the game, and the skip button gives +5 current health. There is a stats panel on the left side of the screen as well as a weapon display and item display that shows your epic, cursed, and legendary items. There is a confirmation window when you select an option and when you hit confirm it updates the stats, changes the colour for increases or decreases, and waits a moment so you can see.

**Comment by PBWiseman** [comment]
Changes:
Fixed the item panel. Now only epic and above rarity items, excluding weapons, will show on it. Other items just change stats so they can be seen in the changes to the stats panel.
Following user feedback, I changed the delay to see stats after selection into having a continue button. This way you can take as long as you want to look at your new stats, while also being able to go past it quicker if you want to.

**Comment by PBWiseman** [comment]
Stat display added to pause menu

**Review by AlexReidNZ — APPROVED** [review]
This new item panel and pause stats menu look really good! Code is clean and readable, ready for merge.

**Review by kblack89 — APPROVED** [review]
This changes and improves how items are selected and is a helpful thing added. No issues that I could find.

File: Assets/Scripts/Items/ItemPanel.cs [hunk 1]
```diff
@@ -258,31 +210,104 @@ private void addItemToList(Item item)
...
+        selectedIndex = -1;
+        selectedItemLabel.text = "Skip";
+        confirmPanel.style.display = DisplayStyle.Flex;
+    }
+
+    private void ConfirmSelection(ClickEvent click)
+    {
+        if (selectedIndex == -1)
+        {
+            PlayerStats.Instance.CurrentHealth += 5;
```
Comment (Loerad) [inline Assets/Scripts/Items/ItemPanel.cs:240]: I got some player testing feedback, they said perhaps changing this to 50% health back for sacrifice of an item. Something we discuss?

**Review by Loerad — CHANGES_REQUESTED** [review]
Overall, really love the new item panel. Way more info than the one I did.  Just some feedback from what I said today.

I think the entire item should be the button to click the item, It should help with clarity. The icons were off centre but I do not know if that is a icon thing or a UI thing that needs a change.

File: Assets/Scripts/Items/ItemPanel.cs (hunk 1 above)
Comment (PBWiseman) [reply Assets/Scripts/Items/ItemPanel.cs]: Fixed

**Review by Loerad — APPROVED** [review]
Looks good!

**Pull request merged**